src/Grids/grid_generation.jl

```diff
@@ -77,12 +88,15 @@ function generate_coordinate(FT, topo::AT, N, H, coord, arch)
     return L, F, C, Δᶠ, Δᶜ
 end
 
-# generate a regularly-spaced coordinate passing the domain extent (2-tuple) and number of points
-function generate_coordinate(FT, topo::AT, N, H, coord::Tuple{<:Number, <:Number}, arch)
+# Generate a regularly-spaced coordinate passing the domain extent (2-tuple) and number of points
+function generate_coordinate(FT, topo::AT, N, H, node_interval::Tuple{<:Number, <:Number}, coordinate_name, arch)
 
-    @assert length(coord) == 2
+    if node_interval[2] < node_interval[1]
+        msg = "$coordinate_name must be an increasing interval!"
+        throw(ArgumentError(msg))
+    end
 
-    c₁, c₂ = @. BigFloat(coord)
+    c₁, c₂ = @. BigFloat(node_interval)
```

Comment:
for 

```Julia
julia> grid = RectilinearGrid(size=3, z= x->x^2, topology=(Flat, Flat, Bounded))
```

here

```Julia
c₁, c₂ = 0, 0
```

is this correct behavior??

Reply:
no?

Reply:
no it's not, e.g., for 

```julia
julia> grid = RectilinearGrid(size=3, z= x->x^2, topology=(Flat, Flat, Bounded))
1×1×3 RectilinearGrid{Float64, Flat, Flat, Bounded} on CPU with 0×0×3 halo
├── Flat x
├── Flat y
└── Bounded  z ∈ [1.0, 16.0]      variably spaced with min(Δz)=3.0, max(Δz)=7.0
```

I'd expect `c₁, c₂ = 1, 16`?


Otherwise if 0, 0 is expected then it's silly to assert that 0 < 0?

Reply:
Nice!